Src/WitsmlExplorer.Frontend/components/QueryEditor.tsx

```diff
@@ -12,56 +12,85 @@ import AceEditor from "react-ace";
 import { useNavigate, useParams } from "react-router-dom";
 import styled from "styled-components";
 import { Colors, dark } from "styles/Colors";
+import React, { FC, useState } from "react";
+import { Chip } from "./StyledComponents/Chip";
+import Icon from "../styles/Icons.tsx";
+import { Stack } from "@mui/material";
 
 export interface QueryEditorProps {
   value: string;
   onChange?: (newValue: string) => void;
+  showCommandPaletteOption?: boolean;
   readonly?: boolean;
 }
 
-export const QueryEditor = (props: QueryEditorProps) => {
-  const { value, onChange, readonly } = props;
+export const QueryEditor: FC<QueryEditorProps> = ({
+  value,
+  onChange,
+  readonly,
+  showCommandPaletteOption
+}) => {
+  const [ace, setAce] = useState<null | any>(null);
+
   const navigate = useNavigate();
   const { serverUrl } = useParams();
   const {
     operationState: { colors }
   } = useOperationState();
 
-  const onLoad = (editor: any) => {
+  const onLoadInternal = (editor: any) => {
     editor.renderer.setPadding(10);
     editor.renderer.setScrollMargin(10);
     if (readonly) {
       editor.renderer.$cursorLayer.element.style.display = "none";
     } else {
       editor.completers = [customCompleter];
+      editor.renderer.on("afterRender", (_: any, renderer: any) => {
```

Comment:
Move the _afterRender_ callback out of the _readonly_ condition. After my previous update, widgets are added even for the readonly editor.

Reply:
Okay, makes sense now